python/ray/air/_internal/checkpoint_manager.py

```diff
@@ -286,6 +286,8 @@ def __init__(
 
         # Used for keeping top K checkpoints.
         self._top_persisted_checkpoints: List[_HeapCheckpointWrapper] = []
+        # Also keep a set of all existing checkpoints
+        self._all_persisted_checkpoint_data: Set[_TrackedCheckpoint] = set()
```

Comment:
We now may schedule multiple saves subsequently (e.g. save is scheduled, trial is instructed to pause which also schedules a save, or maybe PBT schedules an additional save).

Previously, these additional saves were memory checkpoints that were not registered. We want to safeguard here against the same checkpoint being registered multiple times.